### PR TITLE
Dino utils

### PR DESCRIPTION
**Pull request by OmarJabri7**
Utility functions and classes that were added to support DINO as well as their general need to be supported in ivy framework:
- Batch Norm 1D: Added for 1 dimensional batch normalization
- Weight Norm: Added weight normalization since their uses are more common
- weight_norm: utility function for Weight Norm Class
- DINOLoss: class presenting specific loss used in the DINO model

### CODE REVIEW

File: ivy/functional/ivy/experimental/utility.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        Input array
+    out
+        Optional output array, for writing the result to.
+
+    Returns
+    -------
+    ret
+        Input array if it is not None
+    """
+    if x is None:
```
Comment (vedpatwardhan) [inline ivy/functional/ivy/experimental/utility.py:37]: I think `ivy.exists(x)` would be equivalent to this anyway, so this function might be redundant

File: ivy/stateful/layers.py [hunk 2]
```diff
@@ -8,6 +8,26 @@
 
 # ToDo: update docstrings and typehints according to ivy\layers
 
+# Identity #
+# -------#
+class Identity(Module):
```
Comment (vedpatwardhan) [inline ivy/stateful/layers.py:13]: Actually we already have the [`Identity`](https://github.com/unifyai/ivy/blob/8ffd9ffad6f715a3b0bdedb3bbafe52d0085d2e6/ivy/stateful/layers.py#L2199) layer in this file, so we can use that instead

File: ivy/stateful/losses.py [hunk 3]
```diff
@@ -161,3 +161,47 @@ def _forward(
...
+                total_loss += loss.mean()
+                n_loss_terms += 1
+        total_loss /= n_loss_terms
+        self.update_center(teacher_output)
+        return total_loss
+
+    def update_center(self, teacher_output):
+        ivy.stop_gradient(teacher_output)
+        batch_center = ivy.sum(teacher_output, dim=0, keepdim=True)
+        # TODO: Check dist.all_reduce implementation in IVY
```
Comment (vedpatwardhan) [inline ivy/stateful/losses.py:205]: We don't have distributed training supported in `ivy` 😅

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @OmarJabri7, just requested a few changes, thanks 😄

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
